pyproject.toml

```diff
@@ -1,20 +1,87 @@
+[project]
+name = "nipy"
+dynamic = ['version']
+license = {file = "LICENSE"}
+description  = 'A python package for analysis of neuroimaging data'
+readme = 'README.rst'
+classifiers = ["Development Status :: 3 - Alpha",
+               "Environment :: Console",
+               "Intended Audience :: Science/Research",
+               "License :: OSI Approved :: BSD License",
+               "Operating System :: OS Independent",
+               "Programming Language :: Python",
+               "Topic :: Scientific/Engineering"]
+dependencies = [
+    'numpy>=1.22',
+    'scipy>=1.8',
+    'nibabel>=3.2',
+    'sympy>=1.9',
+    'transforms3d'
+]
+
+[[project.maintainers]]
+name = 'nipy developers'
+email = 'neuroimaging@python.org'
+
+[project.urls]
+homepage = 'https://nipy.org/nipy'
+documentation = 'https://scikit-image.org/docs/stable'
+source = 'https://github.com/nipy/nipy'
+download = 'https://pypi.org/project/nipy/#files'
+tracker = 'https://github.com/nipy/nipy/issues'
+
+[project.optional-dependencies]
+build = [
+    'meson-python>=0.13',
+    'wheel',
+    'setuptools>=67',
+    'ninja',
+    'Cython>=3',
+    'numpy>=1.22',
+    'spin==0.3',
+    'build',
+]
+default = [
+    'numpy>=1.22',
+    'scipy>=1.8',
+    'nibabel>=3.2',
+    'sympy>=1.9',
+    'transforms3d'
+]
+developer = [
+    'pre-commit',
+    'rtoml',
+]
+docs = [
+    'sphinx>=7.0',
+    'numpydoc>=1.6.0',
+    'matplotlib',
+    'texext',
+    'ipython'
+]
+optional = [
+    'matplotlib>=3',
+]
+test = [
+    'matplotlib>=3',
+    'pytest>=7.2',
+    'pytest-cov>=4.0',
+    'pytest-doctestplus'
+]
+
 [build-system]
 build-backend = "mesonpy"
 requires = [
   "meson-python>=0.13",
+  "ninja",
   "setuptools",
   "cython>=3",
   # Newer than NEP29-minimum: compile against oldest numpy available
-  "numpy>=1.25; python_version > '3.8'",
+  "numpy>=1.22; python_version > '3.8'",
```

Comment:
@stefanv - am I missing something here?  1.22 is available for Python 3.9.

Reply:
1.25 supports compiling against older APIs.